test/cbmc/proofs/TCP/prvTCPReturnPacket_IPv6/TCPReturnPacket_IPv6_harness.c

```diff
@@ -0,0 +1,179 @@
+                                                                    size_t xNewLength )
+{
+    NetworkBufferDescriptor_t * pxNetworkBuffer = safeMalloc( xNewLength );
+
+    if( ensure_memory_is_valid( pxNetworkBuffer, xNewLength ) )
+    {
+        pxNetworkBuffer->pucEthernetBuffer = safeMalloc( sizeof( TCPPacket_t ) );
+        __CPROVER_assume( pxNetworkBuffer->pucEthernetBuffer );
+
+        /* Add an end point to the network buffer present. Its assumed that the
+         * network interface layer correctly assigns the end point to the generated buffer. */
+        pxNetworkBuffer->pxEndPoint = ( NetworkEndPoint_t * ) malloc( sizeof( NetworkEndPoint_t ) );
+        __CPROVER_assume( pxNetworkBuffer->pxEndPoint != NULL );
+        pxNetworkBuffer->pxEndPoint->pxNext = NULL;
+
+        /* Add an interface */
+        pxNetworkBuffer->pxEndPoint->pxNetworkInterface = ( NetworkInterface_t * ) malloc( sizeof( NetworkInterface_t ) );
+        __CPROVER_assume( pxNetworkBuffer->pxEndPoint->pxNetworkInterface != NULL );
+
+        pxNetworkBuffer->pxEndPoint->pxNetworkInterface->pfOutput = NetworkInterfaceOutputFunction_Stub;
+    }
+
+    return pxNetworkBuffer;
+}
+
+uint16_t usGenerateProtocolChecksum( const uint8_t * const pucEthernetBuffer,
+                                     size_t uxBufferLength,
+                                     BaseType_t xOutgoingPacket )
+{
+    uint16_t usReturn;
+
+    __CPROVER_assert( pucEthernetBuffer != NULL, "The ethernet buffer cannot be NULL" );
+    __CPROVER_assert( __CPROVER_r_ok( pucEthernetBuffer, uxBufferLength ), "pucEthernetBuffer should be readable." );
+
+    return usReturn;
+}
+
+uint16_t usGenerateChecksum( uint16_t usSum,
+                             const uint8_t * pucNextData,
+                             size_t uxByteCount )
+{
+    uint16_t usReturn;
+
+    __CPROVER_assert( pucNextData != NULL, "The next data pointer cannot be NULL" );
+    __CPROVER_assert( __CPROVER_r_ok( pucNextData, uxByteCount ), "The pucNextData should be readable." );
+
+    return usReturn;
+}
+
+void harness()
+{
+    FreeRTOS_Socket_t * pxSocket = ensure_FreeRTOS_Socket_t_is_allocated();
+    NetworkBufferDescriptor_t * pxNetworkBuffer = ensure_FreeRTOS_NetworkBuffer_is_allocated();
+    BaseType_t xReleaseAfterSend;
+    uint32_t ulLen;
+
+    /* The code does not expect both of these to be equal to NULL at the same time. */
+    __CPROVER_assume( pxSocket != NULL || pxNetworkBuffer != NULL );
+
+    /* If network buffer is properly created. */
+    if( ensure_memory_is_valid( pxNetworkBuffer, sizeof( *pxNetworkBuffer ) ) )
+    {
+        /* Assume that the length is proper. */
+        __CPROVER_assume( ( ulLen >= sizeof( TCPPacket_IPv6_t ) ) && ( ulLen < ipconfigNETWORK_MTU ) );
+        pxNetworkBuffer->pucEthernetBuffer = safeMalloc( ulLen + ipSIZE_OF_ETH_HEADER );
+        __CPROVER_assume( pxNetworkBuffer->pucEthernetBuffer );
+
+        pxNetworkBuffer->xDataLength = ( size_t ) ulLen;
```

Comment:
```suggestion
        pxNetworkBuffer->xDataLength = ( size_t )( ulLen + ipSIZE_OF_ETH_HEADER );
```
Why are data length and allocate size different?